bundles/org.openhab.binding.qbus/src/main/java/org/openhab/binding/qbus/internal/QbusBridgeHandler.java

```diff
@@ -0,0 +1,295 @@
+        ScheduledFuture<?> timer = refreshTimer;
+        if (timer != null) {
+            timer.cancel(true);
+            refreshTimer = null;
+        }
+
+        if (refreshInterval == 0) {
+            return;
+        }
+
+        // This timer will check connection with server and client periodically
+        logger.info("Check communication with Server and Client every {} min", refreshInterval);
+        refreshTimer = scheduler.scheduleWithFixedDelay(() -> {
+            logger.info("Checking connection with Qbus Server & Client.");
+
+            QbusCommunication comm = getCommunication();
+
+            if (comm != null) {
+                if (!comm.communicationActive()) {
+                    // Disconnected from Qbus Server, try to reconnect
+                    logger.info("No connection with Qbus Server. Try to restart.");
+                    comm.restartCommunication();
+                    if (!comm.communicationActive()) {
+                        bridgeOffline("No connection with Qbus Server");
+                        return;
+                    }
+
+                } else {
+
+                    // Controller disconnected from Qbus client, try to reconnect controller
+                    if (!comm.clientConnected()) {
+                        logger.info("No connection with Qbus Client. Try to restart.");
+                        comm.restartCommunication();
+                        if (!comm.clientConnected()) {
+                            bridgeOffline("No connection with Qbus Client");
+                            return;
+                        }
+                    }
+                }
+            }
+
+            logger.info("Connection with Qbus Server & Client is still active.");
+            updateStatus(ThingStatus.ONLINE);
+
+        }, refreshInterval, refreshInterval, TimeUnit.MINUTES);
+    }
+
+    /**
+     * Disposes the Bridge and stops communication with the Qbus server
+     */
+    @Override
+    public void dispose() {
+        ScheduledFuture<?> timer = refreshTimer;
+        if (timer != null) {
+            timer.cancel(true);
+        }
+
+        refreshTimer = null;
+
+        QbusCommunication comm = getCommunication();
+        if (comm != null) {
+            comm.stopCommunication();
+        }
+
+        comm = null;
+    }
+
+    /**
+     * Update the configuration parameters
+     */
+    @Override
+    public void handleConfigurationUpdate(Map<String, Object> configurationParameters) {
+        Configuration configuration = editConfiguration();
+        for (Entry<String, Object> configurationParmeter : configurationParameters.entrySet()) {
+            configuration.put(configurationParmeter.getKey(), configurationParmeter.getValue());
+        }
+        updateConfiguration(configuration);
+        updateStatus(ThingStatus.ONLINE);
+    }
+
+    /**
+     * Sets the configuration prameters
+     */
+    protected synchronized void setConfig() {
```

Comment:
Is there any reason why this is synchronized?

The name `setConfig()` seems a bit misleading. Would `readConfig()` be better?